camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/AbstractJcloudsStubYamlTest.java

```diff
@@ -0,0 +1,179 @@
+            if (launcher != null) launcher.stopServers();
+        }
+    }
+    
+    @Override
+    protected LocalManagementContext newManagementContext() {
+        launcher = new BrooklynCampPlatformLauncherNoServer() {
+            @Override
+            protected LocalManagementContext newMgmtContext() {
+                return AbstractJcloudsStubYamlTest.super.newManagementContext();
+            }
+        };
+        launcher.launch();
+        platform = launcher.getCampPlatform();
+        LocalManagementContext mgmt = (LocalManagementContext) launcher.getBrooklynMgmt();
+        return mgmt;
+    }
+
+    protected void addStubbedLocationToCatalog() {
+        addCatalogItems(
+                "brooklyn.catalog:",
+                "  id: " + LOCATION_CATALOG_ID,
+                "  version: 1.0.0",
+                "  itemType: location",
+                "  item:",
+                "    type: " + LOCATION_SPEC,
+                "    brooklyn.config:",
+                "      identity: myidentity",
+                "      credential: mycredential",
+                "      jclouds.computeServiceRegistry:",
+                "        $brooklyn:object:",
+                "          type: " + ByonComputeServiceStaticRef.class.getName(),
+                "      " + SshMachineLocation.SSH_TOOL_CLASS.getName() + ": " + RecordingSshTool.class.getName(),
+                "      templateBuilder:",
+                "        $brooklyn:object:",
+                "          type: "+JcloudsStubTemplateBuilder.class.getName(),
+                "          factoryMethod.name: create",
+                "      " + JcloudsLocation.POLL_FOR_FIRST_REACHABLE_ADDRESS_PREDICATE.getName() + ":",
+                "        $brooklyn:object:",
+                "          type: "+Predicates.class.getName(),
+                "          factoryMethod.name: alwaysTrue",
+                "      " + SshMachineLocation.SSH_TOOL_CLASS.getName() + ": " + RecordingSshTool.class.getName(),
+                "      " + WinRmMachineLocation.WINRM_TOOL_CLASS.getName() + ": " + RecordingWinRmTool.class.getName());
+    }
+
+    protected void addCatalogItems(String... catalogYaml) {
+        addCatalogItems(Joiner.on("\n").join(catalogYaml));
+    }
+    
+    protected void addCatalogItems(String catalogYaml) {
+        mgmt().getCatalog().addItems(catalogYaml, false);
+    }
+    
+    protected Entity createAndStartApplication(Reader input) throws Exception {
+        AssemblyTemplate at = platform.pdp().registerDeploymentPlan(input);
+        Assembly assembly;
+        try {
+            assembly = at.getInstantiator().newInstance().instantiate(at, platform);
+        } catch (Exception e) {
+            LOG.warn("Unable to instantiate " + at + " (rethrowing): " + e);
+            throw e;
+        }
+        LOG.info("Test - created " + assembly);
+        final Entity app = mgmt().getEntityManager().getEntity(assembly.getId());
```

Comment:
Not sure I follow. Are you suggesting I can replace this entire method with `return EntityManagementUtils.createStarting(mgmt(), input).get()`? I just tried that, but `testJcloudsInheritanceAndPasswordSecret` then fails because the `EmptySoftwareProcess` doesn't have a location.

I copied this method (verbatim) from `AbstractYamlRebindTest`.